### DISCUSSION TITLE
Delaying Filter List Updates at Startup

### DISCUSSION THREAD

**Discussion by WildByDesign** [AdguardTeam/AdGuardHome#6558] [Q&A] [answered]
When AdGuardHome is first started, it attempts to update filter lists immediately.

Is there any way, via command lines or anything, to delay the filter list updates by 10 seconds or 30 seconds?

**Context:**

I have a situation where `wan` is coming up much later in the boot process which is causing NTP to delay getting proper time for the device.

AGH is starting and binding to the interface correctly, however, filter lists are failing to download because the device does not yet have the proper time.

If I wait another 20-30 seconds and manually update the filter lists, it works. So that is why I am wondering if there is any way to delay filter list updates by a specific amount of seconds.

Thank you.

**Comment by ainar-g** [comment, marked as answer]
This isn't currently possible.  A related feature request is #3044.

I feel like in your configuration your service runner/watcher should wait for NTP before launching AGH.  Something like [this solution for `systemd` v239+](https://raspberrypi.stackexchange.com/a/95195).

**Comment by WildByDesign** [comment]
That is a great idea. Thank you, sir. I will check with the OpenWrt devs and see what the options are with init scripts.
